torch/csrc/jit/mobile/model_compatibility.h

```diff
@@ -53,6 +53,7 @@ TORCH_API std::unordered_map<std::string, OperatorInfo> _get_model_ops_and_info(
 struct ModelCompatibilityInfo {
   uint64_t bytecode_version;
   std::unordered_map<std::string, OperatorInfo> operator_info;
+  std::vector<at::IValue> type_table;
```

Comment:
Good question! `std::vector<at::IValue>` is the most accurate way to describe type. Each `IValue` will look like `Dict[Int, Tensor]` and there will many different kinds of combinations. However, it's not easy to compare `std::vector<at::IValue>` with the type list `std::unordered_set<std::string>` from runtime. The current way is just to break down the string and extract the token. I'm still not exactly sure the best layer to do the conversion. The api to get type from model hasn't been implemented yet and I just have the logic to do comparison in `is_compatible` api at this moment.

Reply:
I have the same question as @JacobSzwejbka . If there's a type in model of `Dict[Int, Dict[String, custom_class]]`, that is expected to be supported by current code. What would be the type compatibility code work?

Reply:
In latest commit, use `contained_type_` from `type_parser` instead. In the next pr where the custom type is consumed, it should be able to catch this case.

Reply:
With the new implementation (use string instead of Tuple), the following example 
```
"Dict[
   str,
__torch__.A.B.C[
       NamedTuple, [
           [float_features, Tensor],
           [id_list_features, List[Tensor]],
           [label,  Tensor],
           [weight, Tensor],
           ]
       ]
]"
```
will output type set as:
```
(Dict, str, NamedTuple, Tensor, List, Tuple)
```